### DISCUSSION TITLE
[MERGED] Car Proto details missing?

### DISCUSSION THREAD

**Discussion by SarvagyaVaish** [cyberbotics/webots#6463] [General]
I am trying to find more details about the Car proto so that I can create my own car. I see there is a [car-library](https://cyberbotics.com/doc/automobile/car-library?tab-language=python#wbu_car_get_type) with some information about libraries. But, specifically, I am looking for information about the proto.

However, in previous versions of Webots documentation, there was this really helpful [car.md](https://github.com/cyberbotics/webots/blob/d6df3e710e2cd033f614bab976065e732ec91f3f/docs/automobile/car.md) file. I'm curious where I should go to find updated information about the Car proto?

**Comment by omichel** [comment]
[here](https://webots.cloud/run?version=R2023b&url=https://github.com/cyberbotics/webots/blob/released/projects/vehicles/protos/abstract/Car.proto).

**Reply by SarvagyaVaish** [reply]
Interesting. [This](https://webots.cloud/run?url=https://github.com/cyberbotics/webots/tree/released/projects/vehicles/protos/abstract/Car.proto) is what is linked from the docs page [here](https://cyberbotics.com/doc/automobile/proto-nodes), and it's missing all the details.

**Reply by SarvagyaVaish** [reply]
https://webots.cloud/run?url=https://github.com/cyberbotics/webots/tree/released/projects/vehicles/protos/abstract/Car.proto
vs 
https://webots.cloud/run?version=R2023b&url=https://github.com/cyberbotics/webots/blob/released/projects/vehicles/protos/abstract/Car.proto

**Reply by SarvagyaVaish** [reply]
`blob` vs `tree` in the url. Is this work a PR to fix the docs?

**Reply by SarvagyaVaish** [reply]
https://github.com/cyberbotics/webots/tree/released/projects/vehicles/protos/abstract/Car.proto
redirects to 
https://github.com/cyberbotics/webots/blob/released/projects/vehicles/protos/abstract/Car.proto
(aka, the `tree` URL redirects to the `blob` URL)

Perhaps the webots cloud url resolver isn't handling redirects correctly?

https://stackoverflow.com/questions/39400848/in-github-urls-what-is-the-difference-between-a-tree-and-a-blob

**Reply by SarvagyaVaish** [reply]
Nvm.. it's not the resolver. Traced it to this file that defines variables used in the md file. 
https://github.com/cyberbotics/webots/blob/e7fd62944ca05ddcba3153a6583e777e3039565d/docs/js/showdown-extensions.js#L45

Any suggestions? Happy to contribute a fix.

**Reply by omichel** [reply]
I believe `tree` should be replaced with `blob` in case we refer to a file. However, it should remain `tree` if ever we refer to a folder. So probably we should maintain two variables, `github_tree` and `github_blob` are use the first one for folders and the second one for files. Can you propose a PR implementing this?

**Reply by SarvagyaVaish** [reply]
Yeah sure!

**Reply by SarvagyaVaish** [reply]
https://github.com/cyberbotics/webots/pull/6464
Will continue the discussion there.
